transaction/core/src/account_keys.rs

```diff
@@ -108,30 +139,64 @@ impl PublicAddress {
         &self.spend_public_key
     }
 
-    /// Get the optional fog url.
+    /// Get the optional fog url (if it exists / is not empty).
     pub fn fog_url(&self) -> Option<&str> {
-        self.fog_url.as_deref()
+        if self.fog_url.is_empty() {
+            None
+        } else {
+            Some(&self.fog_url)
+        }
+    }
+
+    /// Get the optional fog authority sig (if it exists / is not empty).
+    pub fn fog_authority_sig(&self) -> Option<&[u8]> {
+        if self.fog_authority_sig.is_empty() {
+            None
+        } else {
+            Some(&self.fog_authority_sig)
+        }
+    }
+
+    /// Get the optional fog report key (if it exists / is not empty).
+    pub fn fog_report_key(&self) -> Option<&str> {
+        if self.fog_report_key.is_empty() {
+            None
+        } else {
+            Some(&self.fog_report_key)
+        }
     }
 }
 
 /// Complete AccountKey, containing the pair of secret keys, which can be used
 /// for spending, and optionally some fog-related info,
 /// can be used for spending. This should only ever be present in client code.
-#[derive(Clone, Serialize, Deserialize, Message)]
+#[derive(Clone, Message)]
 pub struct AccountKey {
     /// Private key 'a' used for view-key matching.
     #[prost(message, required, tag = "1")]
-    view_private_key: RistrettoPrivate,
+    pub view_private_key: RistrettoPrivate,
```

Comment:
Do these need to be `pub`? Ideally nothing  in this struct should be `pub`, and there should be accessors and constructors.

Reply:
ok -- i guess i started taking the point of view that, `AccountKey` is just plain-old-data and there is no invariant here to protect.

I mean really any values of these members is correct. Any `RistrettoPrivate` value can be the `view_private_key` etc. Any url can be the `fog_url`. Any bytes can be the `fog_authority_key_fingerprint`. Any string can be the `fog_report_key`.

Also like, if there were some invariant that must be enforced in the constructor, it would be bypassed by `prost::Message`.

So stylistically I would say, these should all be pub, and it's okay to treat it as plain old data.

Maybe for public address that's not really the case because the `sig` has some dependency on the keys, and the values would not be correct if the keys or sig changed.

Reply:
I will try removing the `pub` stuff and see what happens, I had thought that `prost::Message` requires the members to be `pub` at one point, but I guess that actually can't be the case

Reply:
i think that worked out fine, lmk what you think